### PR TITLE
fix(datepicker): enhance scrolling behavior for active cell

### PR DESCRIPTION
**Pull request by jason-capsule42**
… view

# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Review Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

<details>
<summary>
RC Checklist
</summary>

### Browsers

[Browsers Support Guide](https://auro.alaskaair.com/support/browsersSupport)

[Dev demo link](https://alaskaairlines.github.io/auro-formkit/)

Android

- [ ] Chrome
- [ ] Firefox

 iOS

- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Desktop

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Scenarios

- [ ] Validated linked issues with issue reporting team
- [ ] Test coverage report review

[Framework playground ](https://github.com/AlaskaAirlines/auro-framework-playground)

- [ ] Next React
- [ ] SvelteKit

</details><br>
**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Improve datepicker calendar scrolling so the active cell itself is brought into view while optimizing arrow-key navigation performance via cached active cell references.

Bug Fixes:
- Ensure scrollToActiveCell scrolls directly to the active date cell, fixing cases where same-month vertical navigation failed to reveal the focused row in mobile fullscreen views.
- Handle scenarios where no rendered cell matches activeCellDate without throwing, making scrolling behavior more robust across edge cases.

Enhancements:
- Introduce caching of the active cell host in the calendar to avoid re-scanning all focusable cells on every scrollToActiveCell call, improving keyboard navigation efficiency.

Tests:
- Add comprehensive tests validating scrollToActiveCell behavior, including correct scrollIntoView options, no-op behavior when no matching cell exists, cache reuse on hot paths, and cache refresh when stale.

### CODE REVIEW

File: components/datepicker/src/styles/color-calendar.scss [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
   &:focus {
     --ds-auro-calendar-nav-btn-border-color: var(--ds-basic-color-brand-primary, #{v.$ds-basic-color-brand-primary});
+    outline: unset;
   }
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply components/datepicker/src/styles/color-calendar.scss]: Focus styling is achieved using other CSS.

File: components/datepicker/src/auro-calendar.js [hunk 2]
```diff
@@ -1130,17 +1150,17 @@ export class AuroCalendar extends RangeDatepicker {
...
             // Same nearest-by-date fallback handleGridKeyDown uses — direct
             // `cells[length - 1]` would land on the end of the new month,
             // far from the user's intended `nextTs`.
             const fallback = this.pickNearestCell(cells, nextTs, 'next');
             if (fallback) {
               this.setActiveCell(fallback.day.date);
-              this.focusActiveCell();
             }
           }
+          this.scrollToActiveCell();
```
Comment (sun-mota) [inline components/datepicker/src/auro-calendar.js:1162]: looks like `this.scrollToActiveCell();` is getting called always with `this.focusActiveCell();`. how about merging this 2 functions or have `his.scrollToActiveCell();` to be called in `this.focusActiveCell();`?

Comment (jason-capsule42) [reply components/datepicker/src/auro-calendar.js:1162]: Maybe,...?

I am worried that might turn into a cascading issue that I wouldn't want to trip over this close to release. Do you see a real need to do this now?

Comment (jason-capsule42) [reply components/datepicker/src/auro-calendar.js:1162]: There are lots of places these two functions are called. I don't know for sure that all places can/should do both.

**Review by jordanjones243 — APPROVED** [review]

**Review by sun-mota — APPROVED** [review]
